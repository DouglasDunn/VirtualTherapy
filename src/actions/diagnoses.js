import uuid from 'uuid';

// ADD_DIAGNOSES
export const addDiagnoses = (diagnoses) => ({
  type: 'ADD_DIAGNOSES',
  diagnoses
});

export const createDiagnoses = (diagnoses) => {
  return (dispatch) => {
    diagnoses = diagnoses
      .split(',')
      .map(diagnose => {
        const id = uuid.v4();
        return {
          id,
          diagnoses: diagnose.trim(),
          medication: "",
          time: ""
        };
      });
    dispatch(addDiagnoses(diagnoses));
  }
};

// ADD_MEDICATION
export const addEditMedication = (id, medicationObject) => ({
  type: 'ADD_EDIT_MEDICATION',
  id,
  medicationObject
});
