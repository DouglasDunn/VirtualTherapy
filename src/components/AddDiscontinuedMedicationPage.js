import React from 'react';

export class AddDiscontinuedMedicationPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      discontinuedMedication: '' ? 'yes' : 'no'
    };
  }

  onDiscontinuedMedicationChange = (e) => {
    const discontinuedMedication = e.target.value;
    this.setState(() => ({ discontinuedMedication }));
  };

  handlePrevious = () => {
    this.props.history.push('/add-medication');
  };

  onSubmit = (e) => {
    e.preventDefault();

    this.props.history.push('/exercise-questions');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Discontinued Medications</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            Have you stopped taking any medications due to side effects?
            <label>
              <input
                type="radio"
                checked={this.state.discontinuedMedication === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiscontinuedMedicationChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.discontinuedMedication === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiscontinuedMedicationChange}
              />
              No
            </label>
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

export default AddDiscontinuedMedicationPage;
