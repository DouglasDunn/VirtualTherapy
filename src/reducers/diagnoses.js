// Diagnoses Reducer

const diagnosesReducerDefaultState = [];

export default (state = diagnosesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DIAGNOSES':
      return [
        ...state,
        action.diagnoses
      ];
    case 'SET_DIAGNOSES':
      return action.diagnoses;
    default:
      return state;
  }
};
