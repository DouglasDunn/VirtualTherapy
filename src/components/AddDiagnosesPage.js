import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
import { startAddDiagnoses } from '../actions/diagnoses';

export class AddDiagnosesPage extends React.Component {
  onSubmit = (diagnoses) => {
    this.props.startAddDiagnoses(diagnoses);
    this.props.history.push('/diagnoses');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

export default connect(undefined, mapDispatchToProps)(AddDiagnosesPage);
