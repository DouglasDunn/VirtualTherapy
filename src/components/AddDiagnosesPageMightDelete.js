import React from 'react';
import { connect } from 'react-redux';
import DiagnosesForm from './DiagnosesForm';
import { startAddDiagnoses } from '../actions/diagnoses';
import DiagnosesListItem from './DiagnosesListItem';

export class AddDiagnosesPage extends React.Component {
  onSubmit = (diagnoses) => {
    this.props.startAddDiagnoses(diagnoses);
    this.props.history.push('/diagnoses');
    console.log('add');
    console.log(this.props);
  };
  // nextPage = () => {
  //   this.props.history.push('/diagnoses');
  // };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <DiagnosesForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

// const mapStateToProps = (state) => ({
//   diagnoses: state.diagnoses
// });

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

// export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);

export default connect(undefined, mapDispatchToProps)(AddDiagnosesPage);

// <div className="content-container">
//   <div className="list-header">
//     <div className="show-for-mobile">Diagnoses</div>
//     <div className="show-for-desktop">Diagnoses</div>
//     <div className="show-for-desktop">Time</div>
//   </div>
//   <div className="list-body">
//     {
//       this.props.diagnoses.length === 0 ? (
//         <div className="list-item list-item--message">
//           <span>No Diagnoses yet, click the button above to add them.</span>
//         </div>
//       ) : (
//         this.props.diagnoses.map((diagnoses) => {
//           return <DiagnosesListItem key={diagnoses.id} {...diagnoses} />;
//         })
//       )
//     }
//   </div>
// </div>
// <div className="content-container">
//   <button onClick={this.nextPage} className="button">Next</button>
// </div>
