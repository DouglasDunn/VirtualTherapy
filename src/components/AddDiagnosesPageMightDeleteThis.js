import React from 'react';
import { connect } from 'react-redux';
import { startAddDiagnoses } from '../actions/diagnoses';

const initialErrorState = {
  diagnosedError: '',
  diagnosesListError: ''
};

export class AddDiagnosesPage extends React.Component {
  constructor(props) {
    super(props);
    let radioButton = 'no';
    let diagnosesArray;
    if (this.props.diagnoses) {
      if (Array.isArray(this.props.diagnoses)) {
        radioButton = 'yes';
        console.log(Array.isArray(this.props.diagnoses));
        console.log(this.props.diagnoses);
        diagnosesArray = this.props.diagnoses.map(({ diagnoses }) => diagnoses).join(', ');
      }
    }

    this.state = {
      diagnosed: this.props.diagnoses ? radioButton: '',
      diagnosesList: this.props.diagnoses ? diagnosesArray : '',
      ...initialErrorState
    };
  }

  onDiagnosedChange = (e) => {
    const diagnosed = e.target.value;
    this.setState(() => ({ diagnosed }));
  };

  onDiagnosesListChange = (e) => {
    const diagnosesList = e.target.value;
    this.setState(() => ({ diagnosesList }));
  };

  validate = () => {
    let diagnosedError = '';
    let diagnosesListError = '';

    if (!this.state.diagnosed) {
      diagnosedError = 'Please choose an option';
    }

    if ((this.state.diagnosed === 'yes') && !this.state.diagnosesList) {
      diagnosesListError = 'diagnoses list cannot be blank';
    } else {
      diagnosesListError = '';
    }

    if (diagnosedError || diagnosesListError) {
      this.setState({ diagnosedError, diagnosesListError });
      return false;
    }

    return true;
  };

  handlePrevious = () => {
    this.props.history.push('/add-drug-allergies');
  };

  onSubmit = (e) => {
    e.preventDefault();
    const isValid = this.validate();
    if (isValid) {
      if (this.state.diagnosed === 'no') {
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: ''
        });
      } else {
        console.log(this.state.diagnosesList);
        this.props.startAddDiagnoses({
          diagnosed: this.state.diagnosed,
          diagnosesList: this.state.diagnosesList
        });
      }
      this.props.history.push('/add-medication');
    }
  };

  renderDiagnosesList = () => {
    if (this.state.diagnosed === 'yes') {
      return (
        <div>
          {this.state.diagnosesListError && <p className="form__error">{this.state.diagnosesListError}</p>}
          <textarea
            placeholder="Type in all your symptoms and diagnoses separated by a comma"
            className="textarea"
            value={this.state.diagnosesList}
            onChange={this.onDiagnosesListChange}
          >
          </textarea>
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Diagnoses or Symptoms</h1>
          </div>
        </div>
        <div className="content-container">
          <form className="form">
            {this.state.diagnosedError && <p className="form__error">{this.state.diagnosedError}</p>}
            Has a doctor ever diagnosed you with a mental health disease?
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'yes'}
                className="radio-input"
                value="yes"
                onChange={this.onDiagnosedChange}
              />
              Yes
            </label>
            <label>
              <input
                type="radio"
                checked={this.state.diagnosed === 'no'}
                className="radio-input"
                value="no"
                onChange={this.onDiagnosedChange}
              />
              No
            </label>
            {this.renderDiagnosesList()}
          </form>
          <div>
            <button onClick={this.handlePrevious} className="button">Previous</button>
            <button onClick={this.onSubmit} className="button">Next</button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  diagnoses: state.diagnoses
});

const mapDispatchToProps = (dispatch) => ({
  startAddDiagnoses: (diagnoses) => dispatch(startAddDiagnoses(diagnoses))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddDiagnosesPage);
