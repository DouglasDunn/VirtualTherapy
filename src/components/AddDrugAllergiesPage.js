import React from 'react';
import { connect } from 'react-redux';
import DrugAllergyForm from './DrugAllergyForm';
import { startAddDrugAllergy } from '../actions/drugAllergy';

export class AddDrugAllergiesPage extends React.Component {
  onSubmit = (drugAllergy) => {
    this.props.startAddDrugAllergy(drugAllergy);
    this.props.history.push('/add-diagnoses');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Drug Allergies</h1>
          </div>
        </div>
        <div className="content-container">
          <DrugAllergyForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddDrugAllergy: (drugAllergy) => dispatch(startAddDrugAllergy(drugAllergy))
});

export default connect(undefined, mapDispatchToProps)(AddDrugAllergiesPage);
