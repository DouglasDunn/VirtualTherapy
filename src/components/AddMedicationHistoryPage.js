import React from 'react';
import { connect } from 'react-redux';
import MedicationHistoryForm from './MedicationHistoryForm';
import { startAddMedicationHistory } from '../actions/medicationHistory';

export class AddMedicationHistoryPage extends React.Component {
  onSubmit = (medicationHistory) => {
    this.props.startAddMedicationHistory(medicationHistory);
    this.props.history.push('/medication-history');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Medication History</h1>
          </div>
        </div>
        <div className="content-container">
          <MedicationHistoryForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddMedicationHistory: (medicationHistory) => dispatch(startAddMedicationHistory(medicationHistory))
});

export default connect(undefined, mapDispatchToProps)(AddMedicationHistoryPage);
