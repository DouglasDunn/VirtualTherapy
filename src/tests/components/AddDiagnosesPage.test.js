import React from 'react';
import { shallow } from 'enzyme';
import { AddDiagnosesPage } from '../../components/AddDiagnosesPage';
import diagnoses from '../fixtures/diagnoses';

let startAddDiagnoses, history, wrapper;

beforeEach(() => {
  startAddDiagnoses = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddDiagnosesPage startAddDiagnoses={startAddDiagnoses} history={history} />);
});

test('should render AddDiagnosesPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('DiagnosesForm').prop('onSubmit')(diagnoses[1]);
  expect(history.push).toHaveBeenLastCalledWith('/diagnoses');
  expect(startAddDiagnoses).toHaveBeenLastCalledWith(diagnoses[1]);
});
