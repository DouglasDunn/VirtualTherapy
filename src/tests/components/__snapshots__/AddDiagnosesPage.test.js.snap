// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddDiagnosesPage correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add Diagnoses or Symptoms
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <DiagnosesForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
